src/hooks/FullRestrictions.sol

```diff
@@ -1,169 +1,35 @@
-            UpdateRestrictionMessageLib.UpdateRestrictionUnfreeze memory m =
-                payload.deserializeUpdateRestrictionUnfreeze();
-            unfreeze(token, m.user.toAddress());
-        } else {
-            revert InvalidUpdate();
-        }
-    }
-
-    /// @inheritdoc IFreezable
-    function freeze(address token, address user) public auth {
-        require(user != address(0), CannotFreezeZeroAddress());
-        require(!root.endorsed(user), EndorsedUserCannotBeFrozen());
-
-        uint128 hookData = uint128(IShareToken(token).hookDataOf(user));
-        IShareToken(token).setHookData(user, bytes16(hookData.setBit(FREEZE_BIT, true)));
-
-        emit Freeze(token, user);
-    }
-
-    /// @inheritdoc IFreezable
-    function unfreeze(address token, address user) public auth {
-        uint128 hookData = uint128(IShareToken(token).hookDataOf(user));
-        IShareToken(token).setHookData(user, bytes16(hookData.setBit(FREEZE_BIT, false)));
-
-        emit Unfreeze(token, user);
-    }
-
-    /// @inheritdoc IFreezable
-    function isFrozen(address token, address user) public view returns (bool) {
-        return uint128(IShareToken(token).hookDataOf(user)).getBit(FREEZE_BIT);
-    }
-
-    /// @inheritdoc IMemberlist
-    function updateMember(address token, address user, uint64 validUntil) public auth {
-        require(block.timestamp <= validUntil, InvalidValidUntil());
-        require(!root.endorsed(user), EndorsedUserCannotBeUpdated());
-
-        uint128 hookData = uint128(validUntil) << 64;
-        hookData.setBit(FREEZE_BIT, isFrozen(token, user));
-        IShareToken(token).setHookData(user, bytes16(hookData));
-
-        emit UpdateMember(token, user, validUntil);
-    }
-
-    /// @inheritdoc IMemberlist
-    function isMember(address token, address user) external view returns (bool isValid, uint64 validUntil) {
-        validUntil = abi.encodePacked(IShareToken(token).hookDataOf(user)).toUint64(0);
-        isValid = validUntil >= block.timestamp;
-    }
-
-    //----------------------------------------------------------------------------------------------
-    // ERC-165
-    //----------------------------------------------------------------------------------------------
-
-    /// @inheritdoc IERC165
-    function supportsInterface(bytes4 interfaceId) external pure override returns (bool) {
-        return interfaceId == type(ITransferHook).interfaceId || interfaceId == type(IERC165).interfaceId;
+        if (isSourceOrTargetFrozen(from, to, hookData)) return false;
+
+        if (isDepositRequest(from, to)) return isTargetMember(to, hookData);
+        if (isDepositFulfillment(from, to)) return true;
+        if (isDepositClaim(from, to)) return isTargetMember(to, hookData);
+        if (isRedeemRequest(from, to)) return isSourceMember(from, hookData);
+        if (isRedeemFulfillment(from, to)) return true;
+        if (isRedeemClaim(from, to)) return isSourceMember(from, hookData);
```

Comment:
This line should be the root for the broken test. AFAICT, in v3 we don't check if the source is actually a member, i.e. with the new API we do
```diff
-        if (isRedeemClaim(from, to)) return isSourceMember(from, hookData);
+        if (isRedeemClaim(from, to)) return true;
````

because we only check the freeze bit on the `fromHookData`: https://github.com/centrifuge/protocol-v3/blob/6f60024a3d49d83a2efc1611756e6ab229c333fc/src/hooks/FullRestrictions.sol#L72-L93